### PR TITLE
Implement vertx-kafka-client instrumentation; single record handler

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Batch handler coming in the next PR

### CODE REVIEW

**Review by laurit — APPROVED** [review]

File: instrumentation/vertx/vertx-kafka-client-3.5/javaagent/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  implementation(project(":instrumentation:kafka:kafka-clients:kafka-clients-common:library"))
+
+  library("io.vertx:vertx-kafka-client:3.5.0")
+
+  // vertx-codegen and vertx-docgen dependencies are needed for Xlint's annotation checking
+  library("io.vertx:vertx-codegen:3.0.0")
+  testLibrary("io.vertx:vertx-docgen:3.0.0")
+
+  // vertx-kafka-client 3.5 uses kafka-clients 0.10.2.1 by default, need to bump it to make instrumentation work
+  testImplementation("org.apache.kafka:kafka-clients:0.11.0.0")
```
Comment (trask) [inline instrumentation/vertx/vertx-kafka-client-3.5/javaagent/build.gradle.kts:26]: I'd support naming this `vertx-kafka-client-3.6` for a simpler documentation/support story, and updating the base test version to 3.6

File: instrumentation/vertx/vertx-kafka-client-3.5/javaagent/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  bootstrap(project(":instrumentation:kafka:kafka-clients:kafka-clients-0.11:bootstrap"))
+  implementation(project(":instrumentation:kafka:kafka-clients:kafka-clients-common:library"))
+
+  library("io.vertx:vertx-kafka-client:3.5.0")
+
+  // vertx-codegen and vertx-docgen dependencies are needed for Xlint's annotation checking
+  library("io.vertx:vertx-codegen:3.0.0")
+  testLibrary("io.vertx:vertx-docgen:3.0.0")
+
+  // vertx-kafka-client 3.5 uses kafka-clients 0.10.2.1 by default, need to bump it to make instrumentation work
```
Comment (trask) [inline instrumentation/vertx/vertx-kafka-client-3.5/javaagent/build.gradle.kts:25]: https://github.com/mateuszrzeszutek/opentelemetry-java-instrumentation/pull/41

Comment (trask) [inline instrumentation/vertx/vertx-kafka-client-3.5/javaagent/build.gradle.kts:25]: https://github.com/mateuszrzeszutek/opentelemetry-java-instrumentation/pull/41

File: instrumentation/vertx/vertx-kafka-client-3.5/javaagent/build.gradle.kts (hunk 1 above)
Comment (mateuszrzeszutek) [reply instrumentation/vertx/vertx-kafka-client-3.5/javaagent/build.gradle.kts]: Done 👍

**Review by trask — APPROVED** [review]

**Pull request merged**